pkg/process/config/config_test.go

```diff
@@ -381,6 +381,60 @@ func TestEnvSiteConfig(t *testing.T) {
 	os.Unsetenv("DD_PROCESS_AGENT_URL")
 }
 
+func TestEnvProcessAdditionalEndpoints(t *testing.T) {
+	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	defer restoreGlobalConfig()
+
+	assert := assert.New(t)
+
+	expected := make(map[string]string)
+	expected["key1"] = "url1.com"
+	expected["key2"] = "url2.com"
+	expected["key3"] = "url2.com"
+	expected["apikey_20"] = "my-process-app.datadoghq.com" // from config file
+
+	os.Setenv("DD_PROCESS_ADDITIONAL_ENDPOINTS", `{"https://url1.com": ["key1"], "https://url2.com": ["key2", "key3"]}`)
+	defer os.Unsetenv("DD_PROCESS_ADDITIONAL_ENDPOINTS")
+
+	agentConfig, err := NewAgentConfig(
+		"test",
+		"./testdata/TestDDAgentConfigYamlAndSystemProbeConfig.yaml",
+		"./testdata/TestDDAgentConfigYamlAndSystemProbeConfig-Net.yaml",
+	)
+	assert.NoError(err)
+
+	for _, actual := range agentConfig.APIEndpoints {
+		assert.Equal(expected[actual.APIKey], actual.Endpoint.Hostname(), actual)
+	}
+}
```

Comment:
nit: could you add a quick test for a malformed env var and make sure the function `loadEnvVariables` doesn't crash and config has no value?

Reply:
Done!